.github/workflows/build_and_test.yml

```diff
@@ -40,6 +40,12 @@ jobs:
       uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.java }}
+    - name: Set maven version to env
+      run: echo "maven_version=$(grep "<maven.version>" "./java/pom.xml" | head -n1 | awk -F '[<>]' '{print $3}')" >> $GITHUB_ENV
+    - name: Set up Maven ${{ env.maven_version }}
+      uses: stCarolas/setup-maven@v4
```

Comment:
Hi, @guiyanakuang . Is it allowed by Apache Software Foundation?
AFS doesn't allow the un-approved GitHub Actions.

Reply:
Oh, I hadn't noticed that, just a moment while I make sure.

Reply:
@dongjoon-hyun I read this link: https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+status.
Learn about security. This pr hands over the installation of maven to a third party, which does pose a significant security risk. 

> NEVER use 3rd-party actions directly in your worfklows - use the "submodule" pattern. Example PR Tobiasz Kędzierski  opened in SuperSet showing how this could be done.

Although I did not find an ASF Approved Action list, I will close this pr first and open it again if I find a safe and approved solution.